src/src/com/microsoft/aad/adal/StorageHelper.java

```diff
@@ -426,36 +421,36 @@ private synchronized KeyPair generateKeyPairFromAndroidKeyStore() throws General
      * Read KeyPair from AndroidKeyStore. 
      */
     private synchronized KeyPair readKeyPair() throws GeneralSecurityException, IOException {
-        if (doesKeyPairExist()) {
-            // Read key pair again
-            Logger.v(TAG, "Reading Key entry");
-            try {
-                KeyStore keyStore = KeyStore.getInstance("AndroidKeyStore");
-                keyStore.load(null);
-                final KeyStore.PrivateKeyEntry entry = (KeyStore.PrivateKeyEntry)keyStore.getEntry(
-                        KEY_STORE_CERT_ALIAS, null);
-                return new KeyPair(entry.getCertificate().getPublicKey(), entry.getPrivateKey());
-            } catch (RuntimeException e) {
-                // There is an issue in android keystore that resets keystore
-                // Issue 61989:  AndroidKeyStore deleted after changing screen lock type
-                // https://code.google.com/p/android/issues/detail?id=61989
-                // in this case getEntry throws
-                // java.lang.RuntimeException: error:0D07207B:asn1 encoding routines:ASN1_get_object:header too long
-                // handle it as regular KeyStoreException
-                ClientAnalytics.logEvent(new AndroidKeyStoreFailureEvent(
-                        new InstrumentationPropertiesBuilder(e)));
-                throw new KeyStoreException(e);
-            }
+        if (!doesKeyPairExist()) {
+            throw new KeyStoreException("KeyPair entry does not exist.");
+        }
+
+        Logger.v(TAG, "Reading Key entry");
+        final KeyStore keyStore = KeyStore.getInstance("AndroidKeyStore");
+        keyStore.load(null);
+
+        try {
+            final KeyStore.PrivateKeyEntry entry = (KeyStore.PrivateKeyEntry)keyStore.getEntry(
+                    KEY_STORE_CERT_ALIAS, null);
+            return new KeyPair(entry.getCertificate().getPublicKey(), entry.getPrivateKey());
```

Comment:
@weijjia Does this line throw RuntimeException as well?  If so, exactly which api can throw?